lib/mediaType.js

```diff
@@ -99,7 +99,7 @@ function specify(type, spec) {
 
   return {
     q: spec.q,
-    i: spec.i,
+    i: order,
```

Comment:
This breaking keeping in client-specified order and instead replaces it with server-specified order. I think the server order makes sense in certain situations, but this it's the correct fix for it. If you can update the fix, that would be great, otherwise I'll look into it.